pkg/controller/multiclusterhub/common.go

```diff
@@ -685,3 +694,82 @@ func getDeploymentByName(allDeps []*appsv1.Deployment, desiredDeploy string) (*a
 	}
 	return nil, false
 }
+
+func addProxyEnvVarsToDeployment(dep *appsv1.Deployment) *appsv1.Deployment {
+	proxyEnvVars := []corev1.EnvVar{
+		{
+			Name:  "HTTP_PROXY",
+			Value: os.Getenv("HTTP_PROXY"),
+		},
+		{
+			Name:  "HTTPS_PROXY",
+			Value: os.Getenv("HTTPS_PROXY"),
+		},
+		{
+			Name:  "NO_PROXY",
+			Value: os.Getenv("NO_PROXY"),
+		},
+	}
+	for i := 0; i < len(dep.Spec.Template.Spec.Containers); i++ {
+		dep.Spec.Template.Spec.Containers[i].Env = append(dep.Spec.Template.Spec.Containers[i].Env, proxyEnvVars...)
+	}
+
+	return dep
+}
+
+func addProxyEnvVarsToSub(u *unstructured.Unstructured) *unstructured.Unstructured {
+	path := "spec.packageOverrides[].packageOverrides[].value.hubconfig."
+
+	sub, err := injectMapIntoUnstructured(u.Object, path, map[string]interface{}{
```

Comment:
The unstructured package has a few helper functions for working with unstructured resources as well. [SetNestedMap](https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1/unstructured#SetNestedMap) probably follows a similar technique for deeply nested stuff

Reply:
@JakobGray This wont work here bc - 
```
value cannot be set because .spec.packageOverrides is not a map[string]interface{}
``` 
It is of type - `[]map[string]interface{}`